File: project/app/src/main/java/org/owntracks/android/ui/regions/RegionsViewModel.java
package org.owntracks.android.ui.regions;

import org.owntracks.android.data.WaypointModel;
import org.owntracks.android.data.repos.WaypointsRepo;
import org.owntracks.android.services.LocationProcessor;
import org.owntracks.android.ui.base.viewmodel.BaseViewModel;

import javax.inject.Inject;

import dagger.hilt.android.scopes.ActivityScoped;
import io.objectbox.query.Query;
import timber.log.Timber;

@ActivityScoped
public class RegionsViewModel extends BaseViewModel<RegionsMvvm.View> implements RegionsMvvm.ViewModel<RegionsMvvm.View> {
    private final LocationProcessor locationProcessor;
    private final WaypointsRepo waypointsRepo;

    @Inject
    public RegionsViewModel(WaypointsRepo waypointsRepo, LocationProcessor locationProcessor) {
        super();
        Timber.v("new vm instantiated");
        this.waypointsRepo = waypointsRepo;
        this.locationProcessor = locationProcessor;
    }

        public Query<WaypointModel> getWaypointsList() {
            return this.waypointsRepo.getAllQuery();
        }

    @Override
    public void delete(WaypointModel model) {
        waypointsRepo.delete(model);
    }

    @Override
    public void exportWaypoints() {
        locationProcessor.publishWaypointsMessage();
    }
}

File: project/app/src/main/java/org/owntracks/android/ui/regions/RegionsViewModel.kt
package org.owntracks.android.ui.regions

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import dagger.hilt.android.lifecycle.HiltViewModel
import javax.inject.Inject
import org.owntracks.android.data.WaypointModel
import org.owntracks.android.data.repos.WaypointsRepo
import org.owntracks.android.services.LocationProcessor

@HiltViewModel
class RegionsViewModel @Inject constructor(
    private val waypointsRepo: WaypointsRepo,
    private val locationProcessor: LocationProcessor
) : ViewModel() {
    val waypointsList: LiveData<List<WaypointModel>> = waypointsRepo.allLive
    fun exportWaypoints() {
        locationProcessor.publishWaypointsMessage()
    }
}
